### PR TITLE
feat: add template for jira endpoints

### PR DESCRIPTION
**Pull request by joshje**
- [X] Ready for review
- [ ] Follows CONTRIBUTING rules
- [ ] Reviewed by @ … (Snyk internal team)

#### What does this PR do?

Adds a template for Jira integration.

#### More info

Docker images will be generated as part of a separate PR.

### CODE REVIEW

**Comment by adrukh** [comment]
Please add a Dockerfile for jira broker client

**Comment by joshje** [comment]
@adrukh This will be added as a separate PR by @odinn1984. This release only includes the templates.

File: client-templates/jira/.env.sample [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+# your personal username to your Jira Server account
+JIRA_USERNAME=
+
+# your personal password to your Jira Server account
+JIRA_PASSWORD=
+
+# the host where your Jira Server is running, excluding scheme.
+# i.e. jira.yourdomain.com
+JIRA=
```
Comment (odinn1984) [inline client-templates/jira/.env.sample]: @joshje maybe call it JIRA_URI instead of just JIRA?

Comment (odinn1984) [reply client-templates/jira/.env.sample]: or JIRA_HOST

File: client-templates/jira/accept.json.sample [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+      "path": "/rest/api/2/project/:project",
+      "origin": "https://${JIRA_USERNAME}:${JIRA_PASSWORD}@${JIRA}"
+    },
+    {
+      "//": "used to get createmeta for a jira project & issueType",
+      "method": "GET",
+      "path": "/rest/api/2/issue/createmeta",
+      "origin": "https://${JIRA_USERNAME}:${JIRA_PASSWORD}@${JIRA}",
+    },
+
```
Comment (odinn1984) [inline client-templates/jira/accept.json.sample]: missing a { here?

File: client-templates/jira/.env.sample (hunk 1 above)
Comment (joshje) [reply client-templates/jira/.env.sample]: I added `JIRA_API` as:
```bash
JIRA_API=$JIRA/rest/api/2
```

This follows the same convention as https://github.com/snyk/broker/blob/master/client-templates/bitbucket-server/.env.sample#L15:L18

File: lib/auth-header.js [hunk 3]
```diff
@@ -0,0 +1,14 @@
+const replace = require('./replace-vars');
+
+module.exports = ({scheme, token, username, password}) => {
+  console.log('authHeader', scheme, username);
```
Comment (adrukh) [inline lib/auth-header.js]: 😬

Comment (joshje) [reply lib/auth-header.js]: nothing to see here

**Comment by joshje** [comment]
New since the last review:
- Authorization details can be specified in `accept.json` as a specific object called `auth`, without needing to "bake in" to the URL (was causing issues with special chars in the URL)
- No `JIRA_API` env variable needed (this was a copy paste from GitHub that made it's way to Bitbucket by accident).
- README.md and README.npm.md updated to include updated capabilities

**Pull request merged**